cpp/src/ray/runtime/task/task_spec.h

```diff
@@ -0,0 +1,34 @@
+
+#pragma once
+
+#include <ray/core.h>
+#include <msgpack.hpp>
+#include <vector>
+
+namespace ray {
+namespace api {
+
+class TaskSpec {
```

Comment:
This class has overlaps with `TaskSpecification`. I think we can use that class + a C++ `FunctionDescriptor` class.